Build fails on Macports, argument list too long

## Expected behavior and actual behavior.

**Expected:**  With Macports, `port -v install gdal` should build GDAL from source, and create library files.
**Actual:**  Many sources compile properly.  Then build fails at the major link step `libtool clang++ -o libgdal.la` etc, with `Argument list too long`.

## Steps to reproduce the problem.

1.  On Mac OS, create a custom port directory in user's home directory, such as `~dallured/port/262`.  The user name plus custom directory name should be at least 17 characters long, as shown here.

2.  Install a recent version of Macports, such as 2.6.2, in the custom directory.  Specify custom location using `--prefix=XXX`.  Apply Macports quarantine rules for isolated custom build.  Mainly this means omitting all references to `/usr/local` and `/opt/local` from $PATH and linker environment variables.

3.  `port selfupdate`

4.  `port -v install gdal`

5.  Apply patches and workarounds for known macports build problems, as they arise.  Repeat (4) and (5) until the GDAL source build runs.

## Operating system

Mac OS 10.14.6 (Mojave) 64 bit

## GDAL version and provenance

https://distfiles.macports.org/gdal/gdal-3.0.1.tar.gz

Here is the point of failure from the macports detail build log.  The offending command line is 252781 characters long.  This is slightly over a Mac OS hard limit of a little less than 256K, which can not be easily changed.

There are 1268 arguments to the initial `/bin/sh` command, most of them containing long full paths (only 6 of those shown here).  I added line breaks for clarity, and truncated.  See [the full build log](https://github.com/OSGeo/gdal/files/3909083/main.log) for the original, unmodified long command.
```
:info:build /Library/Developer/CommandLineTools/usr/bin/make libgdal.la
:info:build make[1]: Entering directory `/Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal'
:info:build /bin/sh
        /Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal/libtool
        --mode=link
        --silent
        /usr/bin/clang++
        -std=c++14
        -L/Users/dallured/port/262/lib
        -Wl,-headerpad_max_install_names
        -stdlib=libc++
        -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk
        -arch x86_64
        -lcrypto
        -lqhull
        -lqhull
        -L/Users/dallured/port/262/lib
        -lgeos_c
        -lwebp
        -L/Users/dallured/port/262/lib
        -lsqlite3
        -L/Users/dallured/port/262/lib
        -lexpat
        -lgif
        -lCharLS
        -lpng
        -lzstd
        -L/Users/dallured/port/262/lib/proj6/lib
        -lproj
        -lz
        -lpthread
        -ldl
        -L/usr/local/lib
        -L/Users/dallured/port/262/lib
        -lspatialite
        -lpcre
        -L/Users/dallured/port/262/lib
        -lcurl
        -liconv
        -L/Users/dallured/port/262/lib
        -lxml2
        -L/Users/dallured/port/262/lib
        -lz
        -L/Users/dallured/port/262/lib
        -llzma
        -lpthread
        -liconv
        -L/Users/dallured/port/262/lib
        -licui18n
        -licuuc
        -licudata
        -lm
        -o libgdal.la
        ./ogr/gml2ogrgeometry.lo
        ./ogr/ogr2gmlgeometry.lo
        ./ogr/ogr_api.lo
        ./ogr/ogr_expat.lo
        ./ogr/ogr_fromepsg.lo
        ./ogr/ogr_geo_utils.lo
        ./ogr/ogr_geocoding.lo
        ./ogr/ogr_proj_p.lo
        ./ogr/ogr_srs_dict.lo
        ./ogr/ogr_srs_erm.lo
        ./ogr/ogr_srs_esri.lo
        ./ogr/ogr_srs_ozi.lo
        ./ogr/ogr_srs_panorama.lo
        ./ogr/ogr_srs_pci.lo
        ./ogr/ogr_srs_usgs.lo
        ./ogr/ogr_srs_xml.lo
        ./ogr/ogr_srsnode.lo
        ./ogr/ogr_xerces.lo
        ./ogr/ograpispy.lo
        ./ogr/ograssemblepolygon.lo
        ./ogr/ogrcircularstring.lo
        ./ogr/ogrcompoundcurve.lo
        ./ogr/ogrct.lo
        ./ogr/ogrcurve.lo
        ./ogr/ogrcurvecollection.lo
        ./ogr/ogrcurvepolygon.lo
        ./ogr/ogrfeature.lo
        ./ogr/ogrfeaturedefn.lo
        ./ogr/ogrfeaturequery.lo
        ./ogr/ogrfeaturestyle.lo
        ./ogr/ogrfielddefn.lo
        ./ogr/ogrgeomediageometry.lo
        ./ogr/ogrgeometry.lo
        ./ogr/ogrgeometrycollection.lo
        ./ogr/ogrgeometryfactory.lo
        ./ogr/ogrgeomfielddefn.lo
        ./ogr/ogrlinearring.lo
        ./ogr/ogrlinestring.lo
        ./ogr/ogrmulticurve.lo
        ./ogr/ogrmultilinestring.lo
        ./ogr/ogrmultipoint.lo
        ./ogr/ogrmultipolygon.lo
        ./ogr/ogrmultisurface.lo
        ./ogr/ogrpgeogeometry.lo
        ./ogr/ogrpoint.lo
        ./ogr/ogrpolygon.lo
        ./ogr/ogrpolyhedralsurface.lo
        ./ogr/ogrspatialreference.lo
        ./ogr/ogrsurface.lo
        ./ogr/ogrtriangle.lo
        ./ogr/ogrtriangulatedsurface.lo
        ./ogr/ogrutils.lo
        ./ogr/swq.lo
        ./ogr/swq_expr_node.lo
        ./ogr/swq_op_general.lo
        ./ogr/swq_op_registrar.lo
        ./ogr/swq_parser.lo
        ./ogr/swq_select.lo
        /Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal/alg/contour.lo
        /Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal/alg/delaunay.lo
        /Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal/alg/gdal_crs.lo
<omitted 1147 similar args with same long prefix>
        /Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal/third_party/o/Lerc2.lo
        /Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal/third_party/o/Lerc_c_api_impl.lo
        /Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal/third_party/o/RLE.lo
        -rpath /Users/dallured/port/262/lib
        -no-undefined
        -version-info 26:1:0
:info:build make[1]: /bin/sh: Argument list too long
:info:build make[1]: *** [libgdal.la] Error 1
:info:build make[1]: Leaving directory `/Volumes/User/dallured/port/262/var/macports/build/_Volumes_User_dallured_port_262_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_gis_gdal/gdal/work/gdal-3.0.1/gdal'
:info:build make: *** [check-lib] Error 2
```

--- a/gdal/GNUmakefile
+++ b/gdal/GNUmakefile
@@ -1,30 +1,30 @@
 include GDALmake.opt
 
-GDAL_OBJ	=	$(GDAL_ROOT)/frmts/o/*.o \
-			$(GDAL_ROOT)/gcore/*.o \
-			$(GDAL_ROOT)/port/*.o \
-			$(GDAL_ROOT)/alg/*.o \
-			$(GDAL_ROOT)/apps/commonutils.o \
-			$(GDAL_ROOT)/apps/gdalinfo_lib.o \
-			$(GDAL_ROOT)/apps/gdalmdiminfo_lib.o \
-			$(GDAL_ROOT)/apps/gdal_translate_lib.o \
-			$(GDAL_ROOT)/apps/gdalmdimtranslate_lib.o \
-			$(GDAL_ROOT)/apps/gdalwarp_lib.o \
-			$(GDAL_ROOT)/apps/ogr2ogr_lib.o \
-			$(GDAL_ROOT)/apps/gdaldem_lib.o \
-			$(GDAL_ROOT)/apps/nearblack_lib.o \
-			$(GDAL_ROOT)/apps/gdal_grid_lib.o \
-			$(GDAL_ROOT)/apps/gdal_rasterize_lib.o \
-			$(GDAL_ROOT)/apps/gdalbuildvrt_lib.o
+GDAL_OBJ	=	frmts/o/*.o \
+			gcore/*.o \
+			port/*.o \
+			alg/*.o \
+			apps/commonutils.o \
+			apps/gdalinfo_lib.o \
+			apps/gdalmdiminfo_lib.o \
+			apps/gdal_translate_lib.o \
+			apps/gdalmdimtranslate_lib.o \
+			apps/gdalwarp_lib.o \
+			apps/ogr2ogr_lib.o \
+			apps/gdaldem_lib.o \
+			apps/nearblack_lib.o \
+			apps/gdal_grid_lib.o \
+			apps/gdal_rasterize_lib.o \
+			apps/gdalbuildvrt_lib.o
 
-GDAL_OBJ += $(GDAL_ROOT)/ogr/ogrsf_frmts/o/*.o
+GDAL_OBJ += ogr/ogrsf_frmts/o/*.o
 
 ifeq ($(GNM_ENABLED),yes)
-   GDAL_OBJ += $(GDAL_ROOT)/gnm/*.o $(GDAL_ROOT)/gnm/gnm_frmts/o/*.o
+   GDAL_OBJ += gnm/*.o gnm/gnm_frmts/o/*.o
 endif
 
 ifeq ($(HAVE_LERC),yes)
-   GDAL_OBJ += $(GDAL_ROOT)/third_party/o/*.o
+   GDAL_OBJ += third_party/o/*.o
 endif
 
 include ./ogr/file.lst

